pyanaconda/modules/network/nm_client.py

```diff
@@ -0,0 +1,667 @@
+                return True
+            return False
+        else:
+            device_name = device_name or interface_name
+            if not device_name:
+                log.warning("%s no device to look for mac", msg)
+                return False
+            device = nm_client.get_device_by_iface(device_name)
+            if device:
+                hwaddr = device.get_permanent_hw_address() or device.get_hw_address()
+                if interface_name and bind_exclusively:
+                    connection.get_setting_connection().props.interface_name = None
+                    log.debug("%s removed interface-name from connection", msg)
+                s_wired.props.mac_address = hwaddr
+                log.debug("%s bound to %s", msg, hwaddr)
+                return True
+    # bind to device name
+    else:
+        if device_name:
+            if mac_address and bind_exclusively:
+                s_wired.props.mac_address = None
+                log.debug("%s removed mac-address from connection", msg)
+            s_con.props.interface_name = device_name
+            log.debug("%s %s -> %s", msg, interface_name, device_name)
+            return True
+        else:
+            if not interface_name:
+                log.debug("%s no device to bind to", msg)
+                return False
+            else:
+                log.debug("%s already bound to %s", msg, interface_name)
+                if mac_address and bind_exclusively:
+                    s_wired.props.mac_address = None
+                    log.debug("%s removed mac-address from connection", msg)
+                    return True
+                return False
+
+
+def ensure_active_connection_for_device(nm_client, uuid, device_name, only_replace=False):
+    """Make sure active connection of a device is the one specified by uuid.
+
+    :param uuid: uuid of the connection to be applied
+    :type uuid: str
+    :param device_name: name of device to apply the connection to
+    :type device_name: str
+    :param only_replace: apply the connection only if the device has different
+                            active connection
+    :type only_replace: bool
+    """
+    activated = False
+    active_uuid = None
+    device = nm_client.get_device_by_iface(device_name)
+    if device:
+        ac = device.get_active_connection()
+        if ac or not only_replace:
+            active_uuid = ac.get_uuid()
+            if uuid != active_uuid:
+                ifcfg_con = nm_client.get_connection_by_uuid(uuid)
+                # TODO make the API calls synchronous ?
+                nm_client.activate_connection_async(ifcfg_con, None, None, None)
+                activated = True
+    msg = "activated" if activated else "not activated"
+    log.debug("ensure active ifcfg connection for %s (%s -> %s): %s",
+                device_name, active_uuid, uuid, msg)
+    return activated
+
+def update_iface_setting_values(nm_client, iface, new_values):
```

Comment:
Missing 2 empty lines before this top level function and the functions below.

Reply:
https://github.com/rhinstaller/anaconda/pull/1771/commits/ba752145484a43443c54981bbcfad4dc36696a16